server/src/routes/api.js

```diff
@@ -0,0 +1,181 @@
+    const querySQL = `select
+                        posts.post_id,
+                        posts.title,
+                        posts.link,
+                        posts.description,
+                        posts.votes, 
+                        json_agg(json_build_object('tags_id', tags.tag_id, 'tags_title', tags.title)) as tags
+                      from
+                        posts
+                        left outer join post_tags on post_tags.post_id = posts.post_id
+                        left outer join tags on tags.tag_id = post_tags.tag_id
+                      where 
+                        category_id = $1
+                      group by
+	                      posts.post_id
+                      order by
+                        posts.post_id asc;`;
+
+      database.query(querySQL, [req.params.categoryId]).then((response) => {
+      res.json(response.rows);
+    }).catch((error) => {
+      res.status(500).json({error})
+    })
+  });
+
+  router.get('/weeks', (req, res) => {
+    const querySQL = `select
+                        weeks.week_id,
+                        weeks.title,
+                        json_agg(json_build_object('category_id', categories.category_id, 'category_title', categories.title)) as categories
+                      from
+                        weeks
+                        inner join categories on categories.week_id = weeks.week_id
+                      group by
+                        weeks.week_id
+                      order by
+                        weeks.title asc;`;
+
+    database.query(querySQL, [])
+      .then((response) => {
+        res.json(response.rows);
+      }).catch((error) => {
+        res.status(500).json({error})
+      })
+  })
+
+  router.get('/categories', (req, res) => {
+    const querySQL = `select
+                      category_id,
+                      title
+                    from
+                      categories
+                    order by
+                      title asc;`;
+
+    database.query(querySQL, [])
+      .then((response) => {
+        res.json(response.rows);
+      }).catch((error) => {
+        res.status(500).json({error})
+      })
+  })
+
+  router.post('/votes', (req, res) => {
+    //console.log(req.body);
+
+    const { userId, postId } = req.body.vote
+
+    //console.log(userId, postId);
+
+    // attempt to insert vote into user post vote
+    const queryInsertVote = `insert into
+                                user_post_votes
+                                  (
+                                    user_id,
+                                    post_id
+                                  )
+                                  values
+                                  (
+                                    '${userId}',
+                                    '${postId}'
+                                  );`
+
+    database.query(queryInsertVote, [])
+      .then((response) => {
```

Comment:
You're doing three distinctly different things in this endpoint. I would put each in its own function. If the function returns a promise, you can chain them into something beautiful like: 

```js
function insertVote() {
  const queryInsertVote = '...';
  return database.query(queryInsertVote, []);
}

// Same for update and select vote

router.post('/votes', (req, res) => {
  insertVote()
    .then(updateVote)
    .then(getVoteCount)
    .then(count => res(count));
});
```

Obviously the above code is incomplete, but you get the idea. Separate your code into functions, and use promise chaining for readability.